ldp/graph/common_ops.py

```diff
@@ -201,12 +202,21 @@ def __repr__(self) -> str:
         return super(FxnOp, self).__repr__()
 
 
+class ResponseValidationError(Exception):
+    """Raised when a response from the LLM does not pass user-specified validator."""
+
+
 class LLMCallOp(Op[Message]):
     """An operation for LLM calls interaction."""
 
-    def __init__(self, num_samples_logprob_estimate: int = 0) -> None:
+    def __init__(
+        self,
+        num_samples_logprob_estimate: int = 0,
+        response_validator: Callable[[LLMResult], bool | Awaitable[bool]] | None = None,
```

Comment:
If you want to do this, I think it would be `Callable[[LLMResult], bool] | Callable[[LLMResult], Awaitable[bool]] | None = None`

Unless I am wrong, I don't think `bool | Awaitable[bool]` is the same thing

Reply:
lol, I copied this from you: https://github.com/Future-House/ldp/blob/1ad551a429e295d0c80bce776eeca917442837b6/ldp/alg/algorithms.py#L138 . 

And actually, I think this syntax is correct, see here: https://peps.python.org/pep-0483/#covariance-and-contravariance. Specifically:
> [Callable] is covariant in the return type, but contravariant in the arguments.

Reply:
Okay gotchu and we discussed in person, resolving this one